### DISCUSSION TITLE
How does reverse seeking work?

### DISCUSSION THREAD

**Discussion by telekid** [asciinema/discussions#222] [General]
Howdy! Thanks so much for making asciinema-player open source. I've enjoyed working with it recently.

I was curious as to how reverse seeking was implemented. After a cursory look at the code, it appears that you just [feed each output between the playback head and the target location](https://github.com/asciinema/asciinema-player/blob/669eebc5d32b2db4ecacfcda351853d7a998234f/src/driver/recording.js#L272-L317) (or between the start of the recording and the target location in the case of reverse seeking) to the `vt`. Is that true? I would never have guessed that would be fast enough to scrub backwards from the end of the Star Wars recording. I guess this is a great example of "do the simple thing first."

**Comment by ku1ik** [comment]
Howdy!

Yeah, that's correct, it's done exactly like that. And it's the simplest thing I could do there.

I briefly touch on it in this post https://blog.asciinema.org/post/smaller-faster/

There's embedded player in that post, which is high bandwidth one, lots of color sequences etc, and it's equally fast to reverse-seek on it as on the Star Wars one. Rust/WASM is just super fast.

**Reply by telekid** [reply]
Awesome. Thanks for confirming!
